src/fondant/core/manifest.py

```diff
@@ -196,13 +143,41 @@ def copy(self) -> "Manifest":
     def metadata(self) -> t.Dict[str, t.Any]:
         return self._specification["metadata"]
 
+    @property
+    def index(self) -> t.Dict[str, t.Any]:
+        return self._specification["index"]
+
     def update_metadata(self, key: str, value: t.Any) -> None:
         self.metadata[key] = value
 
     @property
     def base_path(self) -> str:
         return self.metadata["base_path"]
 
+    @property
+    def field_mapping(self):
```

Comment:
I was adding some comments to this method, but see you updated it in #654. Do you think we can move the changes here? It should be possible with some git magic.

My comments:
- [x] Add a return type
- [x] Sorted with index first
- [ ] Include index 

Most of them seem to be resolved, except for the last one. I think we need to explicitly include the index, since only the index might be written by a filter component.

Reply:
Thanks!